### PR TITLE
[JENKINS-54599] - Upgrade the plugin from 9.4.5.v20170502 to 9.4.12.v20180830

### PR DESCRIPTION
**Pull request by oleg-nenashev**
https://issues.jenkins-ci.org/browse/JENKINS-54599

- [x] Upgrade the plugin from 9.4.5.v20170502 to 9.4.12.v20180830
- [x] Add Integration tests from the upstream

It may work or not, but looks promising. All relevant integration tests seem to pass

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@jenkinsci/java11-support

**Review by alecharp — APPROVED** [review]
not sure I could spot a problem, but build seems ok.

**Comment by oleg-nenashev** [comment]
At least we have got some integration tests

**Review by olamy — APPROVED** [review]
LGTM

File: pom.xml [hunk 1]
```diff
@@ -2,19 +2,22 @@
...
-  <name>Maven Jenkins Dev Plugin (Jetty)</name>
+  <name>Jenkins Dev Plugin (Jetty)</name>
+  <description>Jetty maven plugins. Custom version for Jenkins</description>
   <properties>
-    <mavenVersion>3.5.4</mavenVersion>
-    <pluginToolsVersion>3.5</pluginToolsVersion>
     <bundle-symbolic-name>${project.groupId}.maven.plugin</bundle-symbolic-name>
+    <it.debug>false</it.debug>
+    <jetty.version>9.4.12.v20180830</jetty.version>
+    <jetty.stopKey>FREEBEER</jetty.stopKey>
```
Comment (batmat) [inline pom.xml:19]: :+1: :)

**Review by batmat — APPROVED** [review]
LGTM.


More seriously, the fact dozens of IT pass is probably enough. Also, as this is a dev tool, if ever there's an issue, we'll discover it and fix it.

Thanks for bumping this up Oleg!

File: pom.xml (hunk 1 above)
Comment (olamy) [reply pom.xml:19]: well all the it tests are from Eclipse Jetty.
I definitely need to look at what need to modify from the original plugin. And maybe integrate it in the Jetty code (hopefully I'm committer on the project as well :) )
But honestly all of this copy/paste is a bit ugly...

Comment (batmat) [reply pom.xml:19]: Right, I know but well a free beer is a free beer. :trollface:

**Comment by oleg-nenashev** [comment]
It seems that the permissions in this repo were messed up. Granted permissions to the @jenkinsci/core like it is done in the Maven HPI Plugin.

@olamy would you be able to release it? Only you now have permissions in this repo. If no, would you be fine if I request release permissions for my own and maybe @batmat ?

> More seriously, the fact dozens of IT pass is probably enough. Also, as this is a dev tool, if ever there's an issue, we'll discover it and fix it.

Note that I migrated this ITs from the original repo. Better than nothing, but there is no coverage for Jenkins-specific logic

File: pom.xml (hunk 1 above)
Comment (olamy) [reply pom.xml:19]: :)

**Comment by olamy** [comment]
@oleg-nenashev sure I will. Merging this one as well?

**Comment by olamy** [comment]
well I cannot release neither...
```
    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:2.8.2:deploy (default-deploy) on project maven-jenkins-dev-plugin: Failed to deploy metadata: Could not transfer metadata org.jenkins-ci.tools/maven-metadata.xml from/to maven.jenkins-ci.org (https://repo.jenkins-ci.org/releases/): Access denied to: https://repo.jenkins-ci.org/releases/org/jenkins-ci/tools/maven-metadata.xml -> [Help 1]
```

**Pull request merged**